README.md
**ℹ️ This package is deprecated and no longer maintained. Please use the successor: [homekit-code](https://github.com/simongolms/homekit-code#readme)**

---

# HomeKit QR Code

Generate a pairing HomeKit QR code label for your HomeKit accessory

[![npm version](https://img.shields.io/npm/v/homekit-qrcode.svg)](https://www.npmjs.com/package/homekit-qrcode)
[![Documentation](https://img.shields.io/badge/documentation-yes-brightgreen.svg)](https://github.com/simongolms/homekit-qrcode#readme)
[![Maintenance](https://img.shields.io/badge/Maintained%3F-no-red.svg)](https://github.com/simongolms/homekit-qrcode/graphs/commit-activity)
[![License: MIT](https://img.shields.io/github/license/simongolms/homekit-qrcode)](https://github.com/simongolms/homekit-qrcode/blob/master/LICENSE)
[![Conventional Commits](https://img.shields.io/badge/Conventional%20Commits-1.0.0-yellow.svg)](https://conventionalcommits.org)
[![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/SimonGolms/homekit-qrcode/releases)

## Usage

```sh
npx homekit-qrcode --category=switch --pairingCode=84131633 --setupId=3QYT
```

### CLI

```sh
Usage: homekit-qrcode [options]

Options:
  -h, --help                                           [boolean]
      --version      Show version number               [boolean]
  -c, --category     category of the Homeit accessory  [string] [required] [choices: "other", "bridge", "fan", "garage", "lightbulb", "doorLock", "outlet", "switch", "thermostat", "sensor", "securitySystem", "door", "window", "windowCovering", "programmableSwitch", "rangeExtender", "ipCamera", "videoDoorBell", "airPurifier", "heater", "airConditioner", "humidifier", "dehumidifier", "appleTv", "speaker", "airport", "sprinkler", "faucet", "showerHead", "television", "targetController"]
  -n, --name         name of the generated file        [string] [default: "homekit-qrcode"]
  -o, --output       format of the generated file      [required] [choices: "svg", "png", "jpeg"] [default: "svg"]
  -p, --pairingCode  8 digits pairing code             [string] [required]
  -s, --setupId                                        [string] [required]

Examples:
  npx homekit-qrcode --category=switch --pairingCode=84131633 --setupId=3QYT                             Generate a QR code for a HomeKit switch
  npx homekit-qrcode --category=switch --pairingCode=84131633 --setupId=3QYT --name=switch --output=png  Generate a QR code for a HomeKit switch as switch.png
```

### Output

![qrcode](./docs/qrcode.png)

---

## Local Development

### Install Dependencies

```sh
npm install
```

### Start Development Server

```sh
npm run start -- --category=switch --pairingCode=84131633 --setupId=3QYT
```

### Build

To build `homekit-qrcode` for production, run:

```sh
npm run build
```

Afterwards the executable code is available under `./lib/`.

```sh
cd lib
node index.js --category=switch --pairingCode=84131633 --setupId=3QYT
```

### Run Tests

```sh
npm test
```

### Repair

This command may be useful when obscure errors or issues are encountered. It removes and recreates dependencies of your project.

```sh
npm run repair
```

### Release

Fully automated version management and package publishing via [semantic-release](https://github.com/semantic-release). It bumps the version according to conventional commits, publishes the package to npm and release a new version to GitHub.

#### Automatic Release (GitHub Action) [Recommended]

Make sure that the secrets `GITHUB_TOKEN` and `NPM_TOKEN` are available in GitHub repository.

```sh
npm run release:ci
```

#### Manual Release

Make sure that the environment variables `GITHUB_TOKEN` and `NPM_TOKEN` are set or declared in `.env` and a productive build was previously created via `npm run build`.

```sh
npm run release
```

You can also test the release manually by running the following command:

```sh
npm run release:dry-run
```

---

## FAQ

### Will my HomeKit accessory be able to pair with the generated QR code?

The QR code is not the same as the pairing code, instead it consists of several parameters such as the category, a specific version and other parameters.
This generated setup payload is the actual content of the QR code and can be generated with the [HomeKit Accessory Simulator (HAS)](https://developer.apple.com/documentation/homekit/testing_your_app_with_the_homekit_accessory_simulator) as well.

![qrcode](./docs/apple-homekit-accessory-simulator.png)

### How do I find out the setup id of my HomeKit Accessory?

Scan the QR code with a QR scanner of your choice. You should get a text starting with `X-HM://....` . The next 9 characters are the combination of the different parameters like the pairing code and the category. The remaining characters are the setup id.

```txt
X-HM://0081YCYEP3QYT ◄── Scanned Setup Payload

X-HM:// 008YCYEP 3QYT
  ▲        ▲      ▲
  │        │      └─── Setup ID
  │        └── Combination of Parameters (first 9 characters)
  └─ Starting Content
```

## Author

**Simon Golms**

- Digital Card: `npx simongolms`
- Github: [@SimonGolms](https://github.com/SimonGolms)
- Website: [gol.ms](https://gol.ms)

## Show your support

Give a ⭐️ if this project helped you!

## License

Copyright © 2021 [Simon Golms](https://github.com/simongolms).<br />
This project is [MIT](https://github.com/simongolms/homekit-qrcode/blob/master/LICENSE) licensed.

## Resources

- https://github.com/maximkulkin/esp-homekit
- https://developer.apple.com/documentation/homekit/hmaccessorycategory/accessory_category_types#overview
- https://developer.apple.com/documentation/homekit/testing_your_app_with_the_homekit_accessory_simulator


CHANGELOG.md
# Changelog

## [3.0.2](https://github.com/simongolms/homekit-qrcode/compare/v3.0.1...v3.0.2) (2022-01-28)


### Bug Fixes

* add deprecation information ([549a396](https://github.com/simongolms/homekit-qrcode/commit/549a3961f0d92a49fdad1b75a82e6ce6c143018d))

## [3.0.1](https://github.com/simongolms/homekit-qrcode/compare/v3.0.0...v3.0.1) (2022-01-23)


### Bug Fixes

* **arguments:** typo in category describtion ([c4a9fb9](https://github.com/simongolms/homekit-qrcode/commit/c4a9fb9da93596bf15b42fbb644e3b3bd59ba1f2))

# [3.0.0](https://github.com/simongolms/homekit-qrcode/compare/v2.0.0...v3.0.0) (2021-10-28)


### Features

* **arguments:** add description to option category ([9a4817a](https://github.com/simongolms/homekit-qrcode/commit/9a4817a368edb69b83ddafc68e7ebb6a1b6f2c25))
* **arguments:** add description to option pairingCode ([c8dd482](https://github.com/simongolms/homekit-qrcode/commit/c8dd4824520f9ac95b612444815fae52bbc2c5e1))
* add images as output formats ([9abb6a6](https://github.com/simongolms/homekit-qrcode/commit/9abb6a698c236464a5d88f3474f4e6d1b0f7f953))


### BREAKING CHANGES

* Adjustment of cli arguments. The `--output` parameter will be used to specify the output format (`svg`, `png`, `jpeg`). By default the svg format is used. The `--name` parameter is used to specify the file name. By default the file name `homekit-qrcode` is used.

# [3.0.0-beta.1](https://github.com/simongolms/homekit-qrcode/compare/v2.0.0...v3.0.0-beta.1) (2021-10-28)


### Features

* **arguments:** add description to option category ([9a4817a](https://github.com/simongolms/homekit-qrcode/commit/9a4817a368edb69b83ddafc68e7ebb6a1b6f2c25))
* **arguments:** add description to option pairingCode ([c8dd482](https://github.com/simongolms/homekit-qrcode/commit/c8dd4824520f9ac95b612444815fae52bbc2c5e1))
* add images as output formats ([9abb6a6](https://github.com/simongolms/homekit-qrcode/commit/9abb6a698c236464a5d88f3474f4e6d1b0f7f953))


### BREAKING CHANGES

* Adjustment of cli arguments. The `--output` parameter will be used to specify the output format (`svg`, `png`, `jpeg`). By default the svg format is used. The `--name` parameter is used to specify the file name. By default the file name `homekit-qrcode` is used.

# [2.0.0](https://github.com/simongolms/homekit-qrcode/compare/v1.0.0...v2.0.0) (2021-10-26)


### Features

* update cli argument setup id ([f94ea6f](https://github.com/simongolms/homekit-qrcode/commit/f94ea6f3df690dafc2d4bef780a3c92bcda1ca8d))
* update cli arguments ([c4ea1c4](https://github.com/simongolms/homekit-qrcode/commit/c4ea1c42ca014144c8c2d236b410a778d2513501))


### BREAKING CHANGES

* set setupId to requried
* pairing code accepts only a string of 8 digits

# [2.0.0-beta.2](https://github.com/simongolms/homekit-qrcode/compare/v2.0.0-beta.1...v2.0.0-beta.2) (2021-10-26)


### Features

* update cli argument setup id ([f94ea6f](https://github.com/simongolms/homekit-qrcode/commit/f94ea6f3df690dafc2d4bef780a3c92bcda1ca8d))


### BREAKING CHANGES

* set setupId to requried

# [2.0.0-beta.1](https://github.com/simongolms/homekit-qrcode/compare/v1.0.0...v2.0.0-beta.1) (2021-10-26)


### Features

* update cli arguments ([c4ea1c4](https://github.com/simongolms/homekit-qrcode/commit/c4ea1c42ca014144c8c2d236b410a778d2513501))


### BREAKING CHANGES

* pairing code accepts only a string of 8 digits

# 1.0.0 (2020-12-28)


### Features

* init commit ([7bd3bf2](https://github.com/simongolms/homekit-qrcode/commit/7bd3bf27665cec97f38cac81dd16694aae8e4191))


src/file.ts
import { createCanvas, Image } from 'canvas';
import { OutputFormat } from './utils';

type MimeType = 'image/jpeg' | 'image/png';

const createImage = async (
  svg: string,
  mimetype: MimeType = 'image/jpeg',
  width = 400,
  height = 540,
): Promise<Buffer> => {
  const canvas = createCanvas(width, height);
  const context = canvas.getContext('2d');
  const image = new Image();
  const svgBase64 = `data:image/svg+xml;base64,${Buffer.from(svg, 'binary').toString('base64')}`;

  return new Promise((resolve, reject) => {
    image.onload = () => {
      context.drawImage(image, 0, 0, 400, 540, 0, 0, width, height);
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      resolve(canvas.toBuffer(mimetype as any));
    };

    image.onerror = (error) => {
      console.error(error);
      reject(error);
    };

    image.src = svgBase64;
  });
};

const createSvg = (pairingCode: string, qrCodeSvg: string) => {
  return `<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg viewBox="0 0 400 540" xmlns="http://www.w3.org/2000/svg">
  <title>HomeKit QR Code</title>
  <defs>
    <symbol id="homekit" viewBox="0 0 130 120"><path d="m128.28 49.26-14.16-11.3v-20c0-1.46-.57-1.9-1.6-1.9h-8.94c-1.2 0-1.93.24-1.93 1.9v10L67.81 1.3a4.22 4.22 0 0 0-6.09 0L1.31 49.26c-2.13 1.67-1.53 4.1.83 4.1h11.14v61.1c0 2.77.83 4.34 2.6 5.04a7 7 0 0 0 2.72.5h92.43a7.1 7.1 0 0 0 2.72-.5c1.77-.7 2.6-2.27 2.6-5.03V53.33h11.2c2.26 0 2.86-2.4.73-4.07ZM20.66 48.1a8.45 8.45 0 0 1 3.32-6.97c1.7-1.37 37.24-29.03 38.24-29.83a4.42 4.42 0 0 1 2.66-1.14c1 .07 1.95.47 2.7 1.14l38.2 30a8.43 8.43 0 0 1 3.32 6.96v58.9a4.25 4.25 0 0 1-4.72 4.77H25.05a4.2 4.2 0 0 1-4.39-4.77V48.1Z" fill="#000"/><path d="M37.12 99.03H92.4a3.12 3.12 0 0 0 3.32-3.56v-42.4a5.48 5.48 0 0 0-2.2-5L66.95 26.82c-.58-.5-1.3-.78-2.06-.8-.75.03-1.46.31-2.03.8l-26.6 21.23a5.46 5.46 0 0 0-2.2 5v42.4a3.14 3.14 0 0 0 3.07 3.57Zm4.29-43.17A4.08 4.08 0 0 1 42.97 52l19.95-15.94a2.72 2.72 0 0 1 1.7-.66c.63.02 1.24.25 1.72.66.53.47 19.05 15.1 19.95 15.94a4.07 4.07 0 0 1 1.56 3.86V88.2a2.47 2.47 0 0 1-2.69 2.83H44.1a2.45 2.45 0 0 1-2.7-2.83V55.86Z" fill="#000"/><path d="M53.54 80.67h22.44c1 0 1.73-.34 1.73-1.8V60.73a3.34 3.34 0 0 0-1.23-2.67L65.91 50a1.73 1.73 0 0 0-2.3 0l-10.57 8.13a3.33 3.33 0 0 0-1.23 2.67v18.13c0 1.4.73 1.74 1.73 1.74Zm5.92-17.1a1.3 1.3 0 0 1 .53-1.1l4.3-3.34a.8.8 0 0 1 .96 0s4.12 3.33 4.28 3.33a1.3 1.3 0 0 1 .54 1.1v8.57c0 .6-.3.73-.74.73H60.2c-.4 0-.73 0-.73-.73v-8.57Z"/></symbol>
    <symbol id="0" viewBox="0 0 34 48"><path d="M17 48c11 0 17-9 17-24S28 0 17 0 0 9 0 24s6 24 17 24ZM7 24C7 12 10 6 17 6c4 0 7 3 9 8L7 28v-4Zm10 18c-4 0-7-2-9-8l19-14v4c0 12-3 18-10 18Z"/></symbol>
    <symbol id="1" viewBox="0 0 34 48"><path d="M34 48v-6H21V0h-7L0 9v7l13-9h1v35H0v6h34Z"/></symbol>
    <symbol id="2" viewBox="0 0 34 48"><path d="M0 14.4684V14.664H7.02096V14.4684C7.02096 9.222 10.7987 5.73523 16.501 5.73523C22.1677 5.73523 25.8742 8.86354 25.8742 13.6864C25.8742 17.2383 24.2348 19.7149 17.5702 26.3299L0.356394 43.3401V48H34V42.0692H10.7631V41.5153L22.239 30.3381C30.6143 22.3218 33.1803 18.2811 33.1803 13.3279C33.1803 5.40937 26.5157 0 16.7149 0C6.91405 0 0 5.99593 0 14.4684Z"/></symbol>
    <symbol id="3" viewBox="0 0 34 48"><path d="M11 26h6c6 0 10 3 10 8s-4 8-10 8-10-3-10-7H0c0 8 7 13 17 13s17-6 17-14c0-6-4-10-10-11 5-1 8-5 8-11 0-7-6-12-15-12C7 0 1 5 1 13h6c1-5 4-7 10-7 5 0 8 2 8 7s-3 8-9 8h-5v5Z"/></symbol>
    <symbol id="4" viewBox="0 0 34 48"><path d="M21 48h6V38h7v-6h-7V19h-6v13H7L22 0h-7L0 33v5h21v10Z"/></symbol>
    <symbol id="5" viewBox="0 0 34 48"><path d="M17 48c10 0 17-7 17-16s-6-16-16-16c-4 0-8 2-10 4L9 6h22V0H3L1 27h7c1-3 5-5 9-5 6 0 10 4 10 10s-4 10-10 10c-5 0-10-3-10-8H0c0 8 7 14 17 14Z"/></symbol>
    <symbol id="6" viewBox="0 0 34 48"><path d="M34 32c0-9-6-15-15-15-3 0-6 1-8 3l1-3L25 0h-8L7 14c-5 7-7 12-7 18 0 9 7 16 17 16s17-7 17-16ZM17 42c-6 0-10-4-10-10s4-10 10-10 10 4 10 10-4 10-10 10Z"/></symbol>
    <symbol id="7" viewBox="0 0 34 48"><path d="M4 48h8L34 6V0H0v6h27L4 48Z"/></symbol>
    <symbol id="8" viewBox="0 0 34 48"><path d="M17 48c10 0 17-5 17-13 0-6-4-11-10-12v-1c5-1 8-5 8-10 0-7-6-12-15-12S2 5 2 12c0 5 3 9 8 10v1C4 24 0 29 0 35c0 8 7 13 17 13Zm0-28c-5 0-9-3-9-7 0-5 4-8 9-8s9 3 9 8c0 4-4 7-9 7Zm0 23c-6 0-10-4-10-9s4-8 10-8 10 3 10 8-4 9-10 9Z"/></symbol>
    <symbol id="9" viewBox="0 0 34 48"><path d="M0 16c0 9 6 15 15 15 3 0 6-1 8-3l-1 3L9 48h8l10-14c5-7 7-12 7-18 0-9-7-16-17-16S0 7 0 16ZM17 6c6 0 10 4 10 10s-4 10-10 10S7 22 7 16 11 6 17 6Z"/></symbol>
    <symbol id="qrCode" ${qrCodeSvg.slice(5, -7)}</symbol>
  </defs>
  <rect fill="#000000" height="540" rx="20" width="400"/>
  <rect fill="#ffffff" height="530" rx="15" width="390" x="5" y="5"/>
  <use href="#homekit" height="120" width="130" x="24" y="30"/>
  <use href="#${pairingCode.charAt(0)}" height="48" width="34" x="174" y="30"/>
  <use href="#${pairingCode.charAt(1)}" height="48" width="34" x="228" y="30"/>
  <use href="#${pairingCode.charAt(2)}" height="48" width="34" x="282" y="30"/>
  <use href="#${pairingCode.charAt(3)}" height="48" width="34" x="336" y="30"/>
  <use href="#${pairingCode.charAt(4)}" height="48" width="34" x="174" y="102"/>
  <use href="#${pairingCode.charAt(5)}" height="48" width="34" x="228" y="102"/>
  <use href="#${pairingCode.charAt(6)}" height="48" width="34" x="282" y="102"/>
  <use href="#${pairingCode.charAt(7)}" height="48" width="34" x="336" y="102"/>
  <use href="#qrCode" height="340" width="340" x="30" y="175"/>
</svg>`;
};

export const createFile = async (pairingCode: string, qrCodeSvg: string, output: OutputFormat): Promise<Buffer> => {
  const svg = createSvg(pairingCode, qrCodeSvg);

  if (output === 'svg') {
    return Promise.resolve(Buffer.from(svg, 'binary'));
  }

  return createImage(svg, `image/${output}`);
};


src/index.ts
#!/usr/bin/env node
import fs from 'fs';
import { CATEGORIES } from './homekit';
import { generateQrCodeAsSvg } from './qrcode';
import { createFile } from './file';
import { argv } from './yargs';

const main = async () => {
  const { category, name, output, pairingCode, setupId } = argv;

  const categoryId = CATEGORIES.get(category) as number;
  const qrCodeSvg = await generateQrCodeAsSvg(categoryId, pairingCode, setupId);
  const file = await createFile(pairingCode, qrCodeSvg, output);

  fs.writeFile(`${name}.${output}`, file, 'utf8', (error) => {
    if (error) {
      return console.error(error);
    }
    console.log(
      '\x1b[33m%s\x1b[0m',
      'Thank you for using homekit-qrcode. Unfortunately, this package is deprecated and no longer maintained. Please use the successor:',
    );
    console.log('\x1b[36m%s\x1b[0m', '\n  npx homekit-code qrcode --category=switch --pairingCode=84131633 --setupId=3QYT\n');

    return console.log('HomeKit QR Code successfully generated');
  });
};

main();


src/yargs.ts
import yargs from 'yargs/yargs';
import { CATEGORIES } from './homekit';
import { isNumber, OUTPUT_FORMATS, OUTPUT_FORMAT_DEFAULT } from './utils';

export const argv = yargs(process.argv.slice(2))
  // Attempt to detect the os locale
  .detectLocale(false)
  // Format usage output
  .wrap(null)
  .options({
    category: {
      alias: 'c',
      choices: Array.from(CATEGORIES.keys()),
      describe: 'category of the HomeKit accessory',
      demandOption: true,
      type: 'string',
    },
    help: {
      alias: 'h',
    },
    name: {
      alias: 'n',
      demandOption: false,
      describe: 'name of the generated file',
      default: 'homekit-qrcode',
      type: 'string',
    },
    output: {
      alias: 'o',
      choices: OUTPUT_FORMATS,
      demandOption: true,
      describe: 'format of the generated file',
      default: OUTPUT_FORMAT_DEFAULT,
    },
    pairingCode: {
      alias: 'p',
      demandOption: true,
      describe: '8 digits pairing code',
      type: 'string',
    },
    setupId: {
      alias: 's',
      demandOption: true,
      type: 'string',
    },
  })
  .check(({ pairingCode }) => {
    if (pairingCode.length === 8 && pairingCode.split('').every((char) => isNumber(char))) {
      return true;
    } else {
      throw new Error('Pairing Code must be specified with 8 digits');
    }
  })
  .usage('Usage: homekit-qrcode [options]')
  .example(
    'npx homekit-qrcode --category=switch --pairingCode=84131633 --setupId=3QYT',
    'Generate a QR code for a HomeKit switch',
  )
  .example(
    'npx homekit-qrcode --category=switch --pairingCode=84131633 --setupId=3QYT --name=switch --output=png',
    'Generate a QR code for a HomeKit switch as switch.png',
  )
  .parseSync();


src/homekit.ts
// KUDOS: https://github.com/maximkulkin/esp-homekit/blob/0f3ef2ac2872ffe64dfe4e5d929420af327d48a5/include/homekit/types.h#L41
export const CATEGORIES = new Map([
  ['other', 1],
  ['bridge', 2],
  ['fan', 3],
  ['garage', 4],
  ['lightbulb', 5],
  ['doorLock', 6],
  ['outlet', 7],
  ['switch', 8],
  ['thermostat', 9],
  ['sensor', 10],
  ['securitySystem', 11],
  ['door', 12],
  ['window', 13],
  ['windowCovering', 14],
  ['programmableSwitch', 15],
  ['rangeExtender', 16],
  ['ipCamera', 17],
  ['videoDoorBell', 18],
  ['airPurifier', 19],
  ['heater', 20],
  ['airConditioner', 21],
  ['humidifier', 22],
  ['dehumidifier', 23],
  ['appleTv', 24],
  ['speaker', 26],
  ['airport', 27],
  ['sprinkler', 28],
  ['faucet', 29],
  ['showerHead', 30],
  ['television', 31],
  ['targetController', 32],
]);

// KUDOS: https://github.com/maximkulkin/esp-homekit/blob/0f3ef2ac2872ffe64dfe4e5d929420af327d48a5/tools/gen_qrcode#L18
export const createSetupUri = (
  category: number,
  password: string,
  setupId: string,
  version = 0,
  reserved = 0,
  flags = 2,
): string => {
  let payload: bigint | number = 0;
  payload = payload | (version & 0x7);

  payload = payload << 4;
  payload = payload | (reserved & 0xf); // reserved bits

  payload = payload << 8;
  payload = payload | (category & 0xff);

  payload = payload << 4;
  payload = payload | (flags & 0xf);

  payload = BigInt(payload) << BigInt(27);
  payload = BigInt(payload) | BigInt(Number(password) & 0x7fffffff);

  const payloadBase36 = payload.toString(36).toUpperCase().padStart(9, '0');

  return `X-HM://${payloadBase36}${setupId}`;
};


src/category.ts
export const homeKitCategory = new Map([
  ['other', 1],
  ['bridge', 2],
  ['fan', 3],
  ['garage', 4],
  ['lightbulb', 5],
  ['doorLock', 6],
  ['outlet', 7],
  ['switch', 8],
  ['thermostat', 9],
  ['sensor', 10],
  ['securitySystem', 11],
  ['door', 12],
  ['window', 13],
  ['windowCovering', 14],
  ['programmableSwitch', 15],
  ['rangeExtender', 16],
  ['ipCamera', 17],
  ['videoDoorBell', 18],
  ['airPurifier', 19],
  ['heater', 20],
  ['airConditioner', 21],
  ['humidifier', 22],
  ['dehumidifier', 23],
  ['appleTv', 24],
  ['speaker', 26],
  ['airport', 27],
  ['sprinkler', 28],
  ['faucet', 29],
  ['showerHead', 30],
  ['television', 31],
  ['targetController', 32],
]);

export const getHomeKitCategoryId = (category: string | number): number => {
  if (typeof category === 'number') {
    return category;
  }

  return homeKitCategory.get(category.toUpperCase()) || 1;
};
